icu4j/main/classes/core/src/com/ibm/icu/impl/personname/PersonNameFormatterImpl.java

```diff
@@ -128,6 +128,13 @@ public String toString() {
     public String formatToString(PersonName name) {
         // TODO: Should probably return a FormattedPersonName object
 
+        if (!nameScriptMatchesLocale(name, this.locale)) {
```

Comment:
It looks good to me, and I can align the spec with it. 

The hack where we say that ja == zh we can document, and even point to the ticket

Reply:
I'm not sure which ticket you're referring to, but that all sounds fine to me.  As I said, if you've got a better idea, I'm all ears.  Not only is the hack ugly, but it ignores things (like `yue`) that we'd probably also want to treat as the same.

One alternative I thought about was to go back to a more conservative reading of "same script" that would cause `ja` and `zh` to compare as different (because `Jpan` wouldn't match `Hans` or `Hant`), and then a Chinese name formatted in Japanese would just end up using the Chinese formatter.  I don't know if this would have undesirable side effects on other locales, though.  (The different flavors of Chinese might also be a problem, but I'm guessing their patterns are all pretty much the same...)  Of course, we can file a ticket (or maybe that's the ticket you're talking about) and revisit this post 43/73.

Reply:
In https://unicode-org.atlassian.net/browse/CLDR-16457, there is currently proviso for which locales have native vs foreign spacing behavior. I just added another section on having the "use other formatter" be data-driven; needs more work, but that way we could have more customized behavior.

Reply:
Okay.  Sounds good.  (Thanks for the final stamp, by the way!)